lib/tracer/tracer.js

```diff
@@ -89,62 +108,122 @@ Tracer.prototype.event = function (traceInfo, type, data, metaData) {
   }
 
   if (data) {
-    let info = _.pick(traceInfo, 'type', 'name');
+    let info = _.pick(trace, 'type', 'name');
     event.data = this._applyFilters(type, data, info, 'start');
   }
 
-  if (metaData && metaData.name) {
-    event.name = metaData.name;
-  }
-
   if (Kadira.options.eventStackTrace) {
     event.stack = CreateUserStack();
   }
 
-  eventLogger('%s %s', type, traceInfo._id);
+  eventLogger('%s %s', type, trace._id);
 
-  if (lastEvent && !lastEvent.endAt) {
-    if (!lastEvent.nested) {
-      console.error('Monti: invalid trace. Please share the trace below at');
-      console.error('Monti: https://github.com/monti-apm/monti-apm-agent/issues/14');
-      console.dir(traceInfo, { depth: 10 });
-    }
-    let lastNested = lastEvent.nested[lastEvent.nested.length - 1];
-
-    // Only nest one level
-    if (!lastNested || lastNested.endAt) {
-      lastEvent.nested.push(event);
-      return event;
-    }
+  event.level = activeEvent ? activeEvent.level + 1 : 0;
 
+  if (event.level > 1) {
     return false;
   }
 
-  traceInfo.events.push(event);
+  if (activeEvent && !activeEvent.endAt) {
```

Comment:
Are there situations where the active event could end, and there still be more nested events?